graspologic/utils/utils.py

```diff
@@ -701,3 +701,82 @@ def plug_in_variance_estimator(x):
         return covariances
 
     return plug_in_variance_estimator
+
+
+def remove_vertices(graph, indices, return_vertices=False):
+    """
+    Remove vertices from an adjacency matrix.
+
+    Parameters
+    ----------
+    graph: np.ndarray, shape (n, n)
+        Adjacency matrix
+    indices: int or array-like
+        Index/indices of the vertex/vertices to be removed.
+    return_vertices: bool, optional
+        Whether to return the tuple (A, V),
+        where A is the truncated adjacency matrix,
+        V is an array of the removed vertices, by default False
+
+    Returns
+    -------
+    truncated_graph: np.ndarray, shape (k, k), n>k
+        Copy of `graph` without the vertices given by `indices`.
+
+    vertices: np.ndarray or tuple, shape (n_indices, n) (optional)
+        Array of removed vertices without edges to each other.
+        If directed, return a tuple with input and output vertices.
+
+    Examples
+    --------
+    # Undirected
+    >>> A = np.array([[0, 1, 2],
+                      [1, 0, 3],
+                      [2, 3, 0]])
+    >>> remove_vertices(A, 0)
+    array([[0., 3.],
+           [3., 0.]]))
+    >>> remove_vertices(A, 0, return_vertices=True)
+    (array([[0., 3.],
+            [3., 0.]]),
+     array([1., 2.]))
+
+    # Directed
+    >>> B = np.array([[0, 1, 2, 3],
+                      [4, 0, 5, 6],
+                      [7, 8, 0, 9],
+                      [10, 11, 12, 0]])
+    >>> remove_vertices(B, 0, return_vertices=True)
+    (array([[ 0.,  5.,  6.],
+            [ 8.,  0.,  9.],
+            [11., 12.,  0.]]),
+    (array([ 4.,  7., 10.]), array([1., 2., 3.])))
+    >>> remove_vertices(B, [0, -1], return_vertices=True)
+    (array([[0., 5.],
+            [8., 0.]]),
+    (array([[4., 7.],
+            [6., 9.]]),
+    array([[ 1.,  2.],
+            [11., 12.]])))
+    """
+    graph = import_graph(graph)
+    if isinstance(indices, list) and len(indices) >= len(graph):
+        raise IndexError("You must pass in fewer vertex indices than vertices.")
+    if isinstance(indices, int):
+        indices = [indices]
```

Comment:
I'd prefer not to have to do this, but if I don't, 
```
if return_vertices:
    cols = graph[:, indices]
    vertices = cols[mask, :].T  # this line breaks with an IndexError for integer indices
```
I'd like to figure out a cleaner way of dealing with integer indices without having to use `np.squeeze` and converting `indices = n` to `indices = [n]`, but haven't figured out a good way yet

Reply:
got it e214230